shared/params/config.go

```diff
@@ -130,11 +134,15 @@ func BeaconConfig() *BeaconChainConfig {
 // OverrideBeaconConfig(c). Any subsequent calls to params.BeaconConfig() will
 // return this new configuration.
 func OverrideBeaconConfig(c *BeaconChainConfig) {
+	lock.Lock()
+	defer lock.Unlock()
 	beaconConfig = c
 }
 
 // Copy returns a copy of the config object.
 func (c *BeaconChainConfig) Copy() *BeaconChainConfig {
+	lock.Lock()
```

Comment:
This should be RLock?

Reply:
I agree. This is tricker than I thought. Here is the problem: 
```go
func UseToledoConfig() {
	lock.Lock()                                     // We lock here
	defer lock.Unlock()
	beaconConfig = ToledoConfig()
}

func ToledoConfig() *BeaconChainConfig {
	cfg := MainnetConfig().Copy()
	cfg.MinGenesisTime = 1605009600
	return cfg
}

func (c *BeaconChainConfig) Copy() *BeaconChainConfig {
	lock.Lock()                                     // We'll dead lock here 
	defer lock.Unlock()
	config, ok := deepcopy.Copy(*c).(BeaconChainConfig)
	if !ok {
		config = *beaconConfig
	}
	return &config
}
```

Reply:
Ok, fixed. I refactored to the following:

```go
func UseToledoConfig() {
	lock.Lock()                                     // We lock here
	defer lock.Unlock()
	beaconConfig = toledoConfig()
}

// This defines the config for the
// Toledo testnet.
// This is the lock free version.
func toledoConfig() *BeaconChainConfig {
	cfg := MainnetConfig().copy()
	cfg.MinGenesisTime = 1605009600
	cfg.GenesisDelay = 86400
	cfg.GenesisForkVersion = []byte{0x00, 0x70, 0x1E, 0xD0}
	cfg.NetworkName = "toledo"
	cfg.SecondsPerETH1Block = 14
	return cfg
}

// This returns a copy of the config object.
// This is the lock free version.
func (c *BeaconChainConfig) copy() *BeaconChainConfig {
	config, ok := deepcopy.Copy(*c).(BeaconChainConfig)
	if !ok {
		config = *beaconConfig
	}
	return &config
}
```

Please take a look @prestonvanloon